slurm/merge.jl

```diff
@@ -89,42 +95,33 @@ function main(config::Dict)
     mkpath(joinpath(export_dir, "test"))
     mkpath(joinpath(export_dir, "infeasible"))
 
-    for opf in OPFs
-        for file in ["primal", "dual", "meta"]
-            h5open(joinpath(export_dir, opf, "$file.h5"), "r") do f
-                for (idx, split) in zip([train_idx, test_idx, infeasible_idx], ["train", "test", "infeasible"])
-                    mkpath(joinpath(export_dir, split, opf))
-                    h5open(joinpath(export_dir, split, opf, "$file.h5"), "w") do g
-                        for k in keys(f)
-                            val = read(f[k])
-                            g[k] = if k != "config" collect(selectdim(val, ndims(val), idx)) else val end
-                        end
-                    end
-                end
-            end
-            rm(joinpath(export_dir, opf, "$file.h5"))
+    # g[k] = f[k][idx]  ∀ k ∈ keys(f)
+    function _copy_to_new_h5(g, f, idx)
+        for k in keys(f)
+            val = read(f[k])
+            g[k] = collect(selectdim(val, ndims(val), idx))
         end
-        rm(joinpath(export_dir, opf))
     end
-    # same for input
-    h5open(joinpath(export_dir, "input.h5"), "r") do f
-        for (idx, split) in zip([train_idx, test_idx, infeasible_idx], ["train", "test", "infeasible"])
-            h5open(joinpath(export_dir, split, "input.h5"), "w") do g
-                create_group(g, "data")
-                for k in keys(f["data"])
-                    val = read(f["data"][k])
-                    g["data"][k] = collect(selectdim(val, ndims(val), idx))
-                end
-
-                create_group(g, "meta")
-                for k in keys(f["meta"])
-                    val = read(f["meta"][k])
-                    g["meta"][k] = if k != "config" collect(selectdim(val, ndims(val), idx)) else val end
-                end
-            end
+
+    # Split consolidated H5s into train/test/infeasible sets
+    all_h5_paths = ["input"; [joinpath(opf, file) for opf in OPFs for file in ["primal", "dual", "meta"]]]
+    for (idx, split) in zip([train_idx, test_idx, infeasible_idx], ["train", "test", "infeasible"])
+        for p in all_h5_paths
+            src_path = joinpath(export_dir, p * ".h5")
+            h5open(src_path, "r") do f
+                dst_path = joinpath(export_dir, split, p * ".h5")
+                mkpath(dirname(dst_path))
+                h5open(dst_path, "w") do g
+                    _copy_to_new_h5(g, f, idx)
+                end  # close dst
+            end  # close src
         end
     end
```

Comment:
This splits all the input/output data into train/test/infeasible.
Granted, I open/close files more often than technically needed, but it felt more natural to iterate over the splits rather than over the input/output